src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -4984,6 +4985,38 @@ private SourceLocation getSelectionAsSourceLocation(boolean fromClick)
       int column = selPos.getColumn() + 1;
       return SourceLocation.create(file, line, column, fromClick);
    }
+   
+   @Handler
+   void onQuickAddNext()
+   {
+      if (!(docDisplay_ instanceof AceEditor))
```

Comment:
I think the right change might be to add method(s) to DocDisplay so that you don't need the cast.